include/mata/nfa-algorithms.hh

```diff
@@ -35,92 +35,79 @@ namespace Nfa {
 namespace Algorithms {
 
     /**
-     * Complement implemented by determization, adding sink state and making automaton complete. Then it adds
-     * final states which were non final in the original automaton.
-     * @param aut Automaton to be complemented
-     * @param alphabet Alphabet is needed since no symbols needs to be in automaton transitions
-     * @param subset_map Maps states to subsets created during complementation.
-     * @return Complemented automaton
+     * Brzozowski minimization of automata (revert -> determinize -> revert -> determinize).
+     * @param[in] aut Automaton to be minimized.
+     * @return Minimized automaton.
      */
-    Nfa complement_classical(
-            const Nfa&         aut,
-            const Alphabet&    alphabet,
-            std::unordered_map<StateSet, State>* subset_map);
+    Nfa minimize_brzozowski(const Nfa& aut);
 
     /**
-     * Complement implemented by determization and making final states which were non final in the original automaton.
-     * @param aut Automaton to be complemented
-     * @param alphabet Alphabet is needed since no symbols needs to be in automaton transitions
-     * @param params Determines algorithms properties
-     * @param subset_map Maps states to subsets created during complementation.
-     * @return result Complemented automaton
+     * Complement implemented by determization, adding sink state and making automaton complete. Then it adds
+     * final states which were non final in the original automaton.
+     * @param[in] aut Automaton to be complemented.
+     * @param[in] alphabet Alphabet is needed for making the automaton complete.
+     * @param[in] minimize_during_determinization Whether the determinized automaton is computed by (brzozowski) minimization
+     * @return Complemented automaton.
      */
-    Nfa complement_naive(
+    Nfa complement_classical(
             const Nfa&         aut,
             const Alphabet&    alphabet,
-            const StringMap&  params,
-            std::unordered_map<StateSet, State>* subset_map);
+            bool minimize_during_determinization = false);
 
     /**
      * Inclusion implemented by complementation of bigger automaton, intersecting it with smaller and then
      * it checks emptiness of intersection
-     * @param smaller Automaton which language should be included in the bigger one
-     * @param bigger Automaton which language should include the smaller one
-     * @param alphabet Alphabet of the both automaton
-     * @param cex A potential counterexample word which breaks inclusion
+     * @param[in] smaller Automaton which language should be included in the bigger one
+     * @param[in] bigger Automaton which language should include the smaller one
+     * @param[in] alphabet Alphabet of both automata (it is computed automatically, but it is more optimal to set it if you have it)
```

Comment:
Fun fact: `more optimal` does not make sense (except it is somewhat commonly used in computer science, apparently). `optimal` is already the superlative. Something cannot be `more optimal` than something other which is already `optimal`. This is equivalent to saying `more best`. 

To get to the point, however, I would use `more efficient` in this case. `more optimal` gives me the idea of getting better results in regard to their correctness  (smaller automaton in this case).

Reply:
Done